NewDawn/NewDawn/ViewControllers/SettingPageViewControllers/SettingPageViewController.swift

```diff
@@ -20,15 +20,48 @@ class SettingPageViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         ImageUtil.polishCircularImageView(imageView: profileImage)
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(self.imageTapped(gesture:)))// add it to the image view;
+        profileImage.addGestureRecognizer(tapGesture)
+        // make sure imageView can be interacted with by user
+        profileImage.isUserInteractionEnabled = true
+    
         polishUIButton(button: preferenceButton)
         polishUIButton(button: accountButton)
         polishUIButton(button: helpButton)
         polishUIButton(button: settingButton)
-        let dataPath = (NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0] as NSString).appendingPathComponent("PersonalImages/0.jpeg")
-        let img = UIImage(contentsOfFile: dataPath)
-        profileImage.image = img
+        
+        LoginUserUtil.fetchLoginUserProfile() {
```

Comment:
This will download profile image every time you load setting page though. Shouldn't we use document directory to store it as a disk cache?

Reply:
The image by default is automatically cached in a temporary memory cache (see the download() method for details). We can cache it on disk for sure but we also need to wipe them out if we do so.

Reply:
An interesting follow up is to add this disk cache logic to download() method too. So we don't need to worry about image cache on specific usecase.

Reply:
Sure. Personal images can be stored on disk since it is more persistent.